sdks/python/setup.py

```diff
@@ -106,8 +106,7 @@ def get_version():
     'avro>=1.8.1,<2.0.0; python_version < "3.0"',
     'avro-python3>=1.8.1,<2.0.0; python_version >= "3.0"',
     'crcmod>=1.7,<2.0',
-    # Dill doesn't guarantee comatibility between releases within minor version.
-    'dill>=0.3.0,<0.3.1',
+    'dill>=0.3.1.1,<0.4.0',
```

Comment:
Let's restrict this to < 0.3.2 due to possible backwards-incompatibility issues.

Reply:
Also, please keep the comment, and link https://github.com/uqfoundation/dill/issues/347 in the comment. Thank you.

Reply:
Updated. PTAL.